halo2/src/prover.rs

```diff
@@ -119,7 +118,12 @@ impl Halo2Prover {
         };
 
         log::info!("Generating circuit for app snark...");
-        let circuit_app = analyzed_to_circuit(pil, fixed, witness);
+        let (circuit_app, publics) = analyzed_to_circuit(pil, fixed, witness);
+
+        assert_eq!(publics.len(), 1);
+        if !publics[0].is_empty() {
+            unimplemented!("Public inputs are not supported yet");
```

Comment:
I'm not that deep into how the aggregation circuit works and I'm not sure if we have proper tests for it. So I decided to do that in another PR.

Reply:
yea we don't really have structured tests for it, let's do it later. It's tricky to have tests for it because the aggregation circuit is pretty big so we can't run it in our current CI